job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java

```diff
@@ -78,13 +95,19 @@ private boolean shouldIgnore(URIStatus status) {
     AlluxioURI outputDir = new AlluxioURI(config.getOutput());
 
     List<URIStatus> files = Lists.newArrayList();
+    long sum = 0;
     for (URIStatus status : context.getFileSystem().listStatus(inputDir)) {
       if (!shouldIgnore(status)) {
         files.add(status);
+        sum += status.getLength();
       }
     }
     Map<WorkerInfo, ArrayList<CompactTask>> assignments = Maps.newHashMap();
-    int groupSize = Math.max(1, (files.size() + 1) / config.getNumFiles());
+    int numOfFiles = config.getNumFiles();
+    if (numOfFiles == CompactConfig.DYNAMIC_NUM_OF_FILES) {
+      numOfFiles = calcNumOfFiles(files.size(), sum / files.size(), config.getFileSize());
+    }
+    int groupSize = Math.max(1, (files.size() + 1) / numOfFiles);
```

Comment:
How will this constant group size work when the input file sizes are different? Shouldn't the file size configuration be enforced by considering each input file size?